homeassistant/components/habitica/entity.py

```diff
@@ -29,26 +31,84 @@ def __init__(
         self,
         coordinator: HabiticaDataUpdateCoordinator,
         entity_description: EntityDescription,
+        subentry: ConfigSubentry | None = None,
     ) -> None:
         """Initialize a Habitica entity."""
         super().__init__(coordinator)
         if TYPE_CHECKING:
             assert coordinator.config_entry.unique_id
+            assert self.user
         self.entity_description = entity_description
-        self._attr_unique_id = (
-            f"{coordinator.config_entry.unique_id}_{entity_description.key}"
+        self.subentry = subentry
+        unique_id = (
+            subentry.unique_id
+            if subentry is not None and subentry.unique_id
+            else coordinator.config_entry.unique_id
         )
+
+        self._attr_unique_id = f"{unique_id}_{entity_description.key}"
         self._attr_device_info = DeviceInfo(
             entry_type=DeviceEntryType.SERVICE,
             manufacturer=MANUFACTURER,
             model=NAME,
-            name=coordinator.data.user.profile.name,
+            name=self.user.profile.name,
             configuration_url=(
-                URL(coordinator.config_entry.data[CONF_URL])
-                / "profile"
-                / coordinator.config_entry.unique_id
+                URL(coordinator.config_entry.data[CONF_URL]) / "profile" / unique_id
             ),
-            identifiers={(DOMAIN, coordinator.config_entry.unique_id)},
+            identifiers={(DOMAIN, unique_id)},
+        )
+
+        if subentry:
+            self._attr_device_info.update(
+                DeviceInfo(
+                    via_device=(
+                        (
+                            DOMAIN,
+                            f"{coordinator.config_entry.unique_id}_{self.user.party.id!s}",
```

Comment:
This is already a string... (fstring handling).

```suggestion
                            f"{coordinator.config_entry.unique_id}_{self.user.party.id}",
```